pyroute2/fixtures/iproute.py

```diff
@@ -0,0 +1,523 @@
+import errno
+from collections.abc import AsyncGenerator, Generator
+from typing import Union
+
+import pytest
+import pytest_asyncio
+
+from pyroute2 import NDB, netns
+from pyroute2.common import uifname
+from pyroute2.iproute.linux import AsyncIPRoute, IPRoute
+from pyroute2.netlink.exceptions import NetlinkError
+from pyroute2.netlink.rtnl import IFNAMSIZ
+from pyroute2.netlink.rtnl.ifinfmsg import ifinfmsg
+
+
+class TestInterface:
+    '''Test interface spec.
+
+    Provided by `test_link` fixture.
+
+    Provides shortcuts to some important interface properties,
+    like `TestInterface.index` or `TestInterface.netns`.
+    '''
+
+    def __init__(self, index: int, ifname: str, address: str, nsname: str):
+        if index <= 1:
+            raise TypeError('test interface index must be > 1')
+        if not 1 < len(ifname) <= IFNAMSIZ:
+            raise TypeError(
+                'test interface ifname length must be from 2 to IFNAMSIZ'
+            )
+        self._index = index
+        self._ifname = ifname
+        self._address = address
+        self._netns = nsname
+
+    @property
+    def index(self) -> int:
+        '''Test interface index.
+
+        Is always greater than 1, as index 1 has the the loopback interface.
+        '''
+        return self._index
+
+    @property
+    def ifname(self) -> str:
+        '''Test interface ifname.
+
+        The name length must be greater than 1 and less or equal IFNAMSIZ.
+        '''
+        return self._ifname
+
+    @property
+    def address(self) -> str:
+        '''Test interface MAC address.
+
+        In the form `xx:xx:xx:xx:xx:xx`.'''
+        return self._address
+
+    @property
+    def netns(self) -> str:
+        '''Test interface netns.
+
+        A string name of the network namespace.'''
+        return self._netns
+
+
+class TestContext:
+    '''The test context.
+
+    Provided by `async_context` and `sync_context` fixtures.
+
+    Provides convenient shortcuts to RTNL API, the network namespace
+    name and the test interface spec.
+    '''
+
+    def __init__(
+        self, ipr: Union[IPRoute, AsyncIPRoute], test_link: TestInterface
```

Comment:
This fixture class would benefit from being `Generic[IPRouteT]` to avoid confusing the type checker later but i can try to do it later if you want.

Reply:
Done.